tests/common/port_toggle.py

```diff
@@ -1,145 +1,125 @@
+            startup_err_msg = "Some ports did not come up as expected: {}".format(str(down_ports))
     except Exception as e:
-        startup_err_msg = 'Startup interfaces failed with exception: {}'.format(repr(e))
+        startup_err_msg = "Startup ports failed with exception: {}".format(repr(e))
 
     pytest_assert(shutdown_ok, shutdown_err_msg)
     pytest_assert(startup_ok, startup_err_msg)
 
-    logger.info('wait %d seconds for system to startup', wait_after_ports_up)
+    logger.info("Wait %d seconds for system to stabilize", wait_after_ports_up)
     time.sleep(wait_after_ports_up)
 
 
-def get_port_toggle_wait_time(duthost, port_count):
-    """
-    Port toggle wait time depends on many factors: port count, cpu type. This function allow user to customize port
-    toggle wait time according to DUT host information and port count.
-    :param duthost: DUT host object
-    :param port_count: total port count to be toggled
-    :return: port down wait time and port up wait time
+def log_system_resources(duthost, logger):
+    # Watch memory status
+    memory_output = duthost.shell("show system-memory")["stdout"]
+    logger.info("Memory Status: %s", memory_output)
+
+    # Watch orchagent CPU utilization
+    orch_cpu = duthost.shell("show processes cpu | grep orchagent | awk '{print $9}'")["stdout"]
+    logger.info("Orchagent CPU Util: %s", orch_cpu)
+
+    # Watch Redis Memory
+    redis_memory = duthost.shell("redis-cli info memory | grep used_memory_human")["stdout"]
+    logger.info("Redis Memory: %s", redis_memory)
+
+
+def default_port_toggle_wait_time(duthost, port_count):
+    """Get the default timeout for shutting down/starting up a set of ports.
+
+    Port toggle wait time can depend on many factors: port count, cpu type, etc. The callback allows
+    users to customize port toggle wait behavior based on DUT specs and port count.
+
+    Args:
+        duthost: DUT host object
+        port_count: total number of ports to toggle
+
+    Returns
+        (int, int): timeout for shutting down ports, and timeout for bringing up ports
     """
-    port_down_wait_time, port_up_wait_time = 20, 60
+    port_down_wait_time, port_up_wait_time = 120, 180
     asic_type = duthost.facts["asic_type"]
-    if asic_type == 'mellanox':
-        # default t0 topology has 28 ports to toggle
-        base_port_count = 28.0
-        if port_count <= base_port_count:
-            port_count_factor = 1
-        else:
-            port_count_factor = port_count / base_port_count
-        from .mellanox_data import get_chip_type
-        chip_type = get_chip_type(duthost)
-        if chip_type == 'spectrum1':
-            # spectrum based switch usually has less powerful cpu
-            port_down_wait_time = 40
-            port_up_wait_time = 70
-        else:
-            port_down_wait_time = 30
 
+    if asic_type == "mellanox":
+        if port_count <= BASE_PORT_COUNT:
+            port_count = 28.0
```

Comment:
I don't think you need to force type to float for python.

Reply:
Better yet, I will just replace this magic number with the constant we added.